kafka-rest/src/main/java/io/confluent/kafkarest/ProducerMetrics.java

```diff
@@ -0,0 +1,414 @@
+import org.apache.kafka.common.metrics.JmxReporter;
+import org.apache.kafka.common.metrics.KafkaMetricsContext;
+import org.apache.kafka.common.metrics.MeasurableStat;
+import org.apache.kafka.common.metrics.MetricConfig;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Avg;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.CumulativeSum;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.metrics.stats.WindowedCount;
+import org.apache.kafka.common.metrics.stats.WindowedSum;
+import org.apache.kafka.common.utils.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProducerMetrics {
+  public static final String JMX_PREFIX = "kafka.rest";
+
+  private static final Logger log = LoggerFactory.getLogger(ProducerMetrics.class);
+
+  private final Metrics metrics;
+  private final ConcurrentMap<BeanCoordinate, ProduceMetricMBean> beansByCoordinate =
+      new ConcurrentHashMap<>();
+
+  public ProducerMetrics(Time time) {
+    this.metrics = new MetricsBuilder(JMX_PREFIX).withTime(time).build();
+    setupMetricBeans();
+    log.info("Successfully registered kafka-rest produce metrics with JMX");
+  }
+
+  /**
+   * Get or create a {@link ProduceMetricMBean} with the specified name and the given tags. Each
+   * group is uniquely identified by the name and tags.
+   *
+   * @param groupName the name of the ReplicatorMetricGroup group; may not be null
+   * @param tags pairs of tag name and values
+   * @return the {@link ProduceMetricMBean} that can be used to create metrics; never null
+   */
+  public ProduceMetricMBean mbean(String groupName, Map<String, String> tags) {
+    BeanCoordinate beanCoordinate = new BeanCoordinate(groupName, tags);
+    beansByCoordinate.putIfAbsent(beanCoordinate, new ProduceMetricMBean(beanCoordinate));
+    return beansByCoordinate.get(beanCoordinate);
+  }
+
+  /** Stop and unregister the metrics from any reporters. */
+  public void stop() {
+    log.info("Unregistering produce metrics with JMX");
+    metrics.close();
+  }
+
+  /** Sets up a {@link ProduceMetricMBean} for each sensor */
+  private void setupMetricBeans() {
+    ProduceMetricMBean metricMBean =
+        mbean(ProducerMetricsRegistry.GROUP_NAME, Collections.emptyMap());
+
+    // Clear out any previous metrics and sensors that might have been present
+    if (metricMBean != null) {
+      metricMBean.close();
+    }
+
+    // request metrics
+    new SensorBuilder(metricMBean, ProducerMetricsRegistry.REQUEST_SENSOR)
+        .addRate(ProducerMetricsRegistry.REQUEST_RATE, ProducerMetricsRegistry.REQUEST_RATE_DOC)
+        .addCumulativeCount(
```

Comment:
We removed the cumulative counts from rest-utils in the end, and just kept the windowed ones.

Reply:
removed